src/pyclass.rs

```diff
@@ -0,0 +1,642 @@
+        }
+    }
+
+    #[must_use]
+    fn init_class(self, shell: &mut T::ConcreteLayout) -> PyResult<()> {
+        macro_rules! raise_err {
+            ($name: path) => {
+                return Err(PyErr::new::<RuntimeError, _>(format!(
+                    "Base class '{}' is not initialized",
+                    $name
+                )));
+            };
+        }
+        let PyClassInitializer { init, super_init } = self;
+        if let Some(value) = init {
+            unsafe { shell.py_init(value) };
+        } else if T::ConcreteLayout::NEED_INIT {
+            raise_err!(T::NAME);
+        }
+        if let Some(super_init) = super_init {
+            let super_init = unsafe { Box::from_raw(super_init) };
+            if let Some(super_obj) = shell.get_super_or() {
+                super_init.init_class(super_obj)?;
+            }
+        } else if <T::BaseType as PyTypeInfo>::ConcreteLayout::NEED_INIT {
+            raise_err!(T::BaseType::NAME)
+        }
+        Ok(())
+    }
+
+    /// Pass the value that you use in Python to the initializer.
+    pub fn init(&mut self, value: T) {
+        self.init = Some(value);
+    }
+
+    /// Get the initializer for the base object.
+    /// Resembles `super().__init__()` in Python.
+    pub fn get_super(&mut self) -> &mut PyClassInitializer<T::BaseType> {
+        if let Some(super_init) = self.super_init {
+            return unsafe { &mut *super_init };
+        }
+        let super_init = Box::into_raw(Box::new(PyClassInitializer::default()));
+        self.super_init = Some(super_init);
+        unsafe { &mut *super_init }
+    }
+
+    #[doc(hidden)]
+    pub unsafe fn create_shell(self, py: Python) -> PyResult<*mut PyClassShell<T>>
+    where
+        T: PyClass,
+        <T::BaseType as PyTypeInfo>::ConcreteLayout:
+            crate::type_object::PyObjectSizedLayout<T::BaseType>,
+    {
+        let shell = PyClassShell::new(py)?;
+        self.init_class(&mut *shell)?;
+        Ok(shell)
+    }
+}
+
+/// Represets that we can convert the type to `PyClassInitializer`.
+///
+/// It is mainly used in our proc-macro code.
+pub trait IntoInitializer<T: PyClass> {
+    fn into_initializer(self) -> PyResult<PyClassInitializer<T>>;
+}
+
+impl<T: PyClass> IntoInitializer<T> for T {
+    fn into_initializer(self) -> PyResult<PyClassInitializer<T>> {
+        Ok(PyClassInitializer::from_value(self))
+    }
+}
+
+impl<T: PyClass> IntoInitializer<T> for PyResult<T> {
+    fn into_initializer(self) -> PyResult<PyClassInitializer<T>> {
+        self.map(PyClassInitializer::from_value)
+    }
+}
+
+impl<T: PyClass> IntoInitializer<T> for PyClassInitializer<T> {
+    fn into_initializer(self) -> PyResult<PyClassInitializer<T>> {
+        Ok(self)
+    }
+}
+
+impl<T: PyClass> IntoInitializer<T> for PyResult<PyClassInitializer<T>> {
+    fn into_initializer(self) -> PyResult<PyClassInitializer<T>> {
+        self
+    }
+}
+
+/// Register new type in python object system.
```

Comment:
```suggestion
/// Register new type in python's object system.
```

Alternatively you can use "in **the** python object system".